### ISSUE TITLE
Support z.discriminatedUnion for complex schema

### ISSUE THREAD

**Issue by winston0410** [ciscoheat/sveltekit-superforms#197]
**Is your feature request related to a problem? Please describe.**
I am trying to use `z.discriminatedUnion` for a more complex schema. For example a schema like this:

```ts
export const formSchema = z.discriminatedUnion("createProfile", [
  z.object({ createProfile: z.literal(false), email: z.string().email() }),
  z.object({
    createProfile: z.literal(true),
    email: z.string().email(),
    firstName: z.string(),
    lastName: z.string(),
  }),
]);
```

which is more readable than using `refine`. And now there is an error:

```
SuperFormError: Only Zod schema objects can be used with superValidate. Define the schema with z.object({ ... }) and optionally refine/superRefine/transform at the end.
```

**Describe the solution you'd like**
Support `z.discriminatedUnion` in superform

**Describe alternatives you've considered**

**Additional context**

**Comment by ciscoheat** [comment]
This complicates the types a lot unfortunately. Since you can't know at compile-time which of the schemas will be used, you have to merge them to get intellisense, which gets messy in `$errors`, `$tainted`, proxies, and other stores that reflects the form state.

So even if it's possible, it's a lot of work, and it's a rare case so I can't put this high on the priority list, sorry.

**Comment by tobiassern** [comment]
Just in addition to this, the zod team are planning to deprecate disciminatedUnion -> https://github.com/colinhacks/zod/issues/2106

**Comment by winston0410** [comment]
Yes indeed maybe we should revisit this only after `z.switch` is implemented, and keep the rule of only having `z.object` for now.

**Comment by ciscoheat** [comment]
Let's do that, if you can open another issue when it's ready, that would be much appreciated!
